cmd/restic/cmd_copy.go

```diff
@@ -184,57 +185,64 @@ type treeCloner struct {
 	buf          []byte
 }
 
-func (t *treeCloner) copyTree(ctx context.Context, treeID restic.ID) error {
+func (t *treeCloner) copyTree(ctx context.Context, rootTreeID restic.ID) error {
```

Comment:
Now that `copyTree` is no longer recursively called, we could think of calling this with all root tree IDs as argument. But this would need a larger change in the `copy` code and can also be postponed...

Reply:
I've added a commit to remove the `treeCloner` struct for now. Copying all snapshots in one go would be the task of another PR. That would also require some though on the user interface to possibly provide more feedback to the user than is done currently.